### DISCUSSION TITLE
In app messaging - Set up deeplink action in firebase with react-native

### DISCUSSION THREAD

**Discussion by predbjorn** [invertase/react-native-firebase#5883] [Q&A]
Both our deeplinks and in-app-messaging works perfectly in our app. However, I am trying to set up deeplink in Action parameter when sending out a deeplink like this: (see Action (optional) - image action field)

![WvnNr](https://user-images.githubusercontent.com/3714648/143552592-7e90f2dd-bf21-42bb-adb5-a392a66bd3d0.png)
![EV0Sf](https://user-images.githubusercontent.com/3714648/143552616-0581b56a-be0f-4f90-8aa0-d47e36522e87.png)

When I click on the image, I end up in App Store for our app. (I have tested pressing the link separately, which leads me to the correct place in the app).

I use:
```
"@react-native-firebase/in-app-messaging": "^11.5.0",
"react-native": "0.64.0",
```

I cant find any other documentation for how to implement this. Any ideas?

**Comment by mikehardy** [comment]
Hmm - is the a reason not to be on the current versions here? They pull in  newer versions of the SDKs and that *may* solve it. Certainly not worth investigating much until you're definitely on the up to date SDKs.
Dynamic link behavior is quite platform specific - you mention "App Store" so I suppose it's iOS? Is this on Simulator or real device? What version of iOS?
When diagnosing FCM issues it's vital to control the JSON carefully. That is to say, you should craft the JSON manually and use the FCM REST APIs to post the message to your device token, and then you can post the JSON here that demonstrates the problem.
Configuration is also very important - the Info.plist entries are pretty important in this case. You mention it works if you test the link separately but you do not mention any details about that test, so it's hard to reason about that.
Have you tested the link with `?d=1` in a browser to make sure the flows check out there? https://firebase.google.com/docs/dynamic-links/debug

Basically - sounds like there is an issue but there is a lot of related work + information around it before I or anyone can offer help. Hopefully as you work through those something obvious shows up and fixes it for you, otherwise you could post again with status/info after working through that
